group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -10921,6 +10823,1158 @@ public void testLastConsumerProtocolMemberRebalanceTimeoutInConsumerGroup() {
+                    .addTopic(fooTopicId, fooTopicName, 2)
+                    .addRacks()
+                    .build())
+                .withConsumerGroup(new ConsumerGroupBuilder(groupId, 10)
+                    .withSubscriptionMetadata(new HashMap<String, TopicMetadata>() {
+                        {
+                            put(fooTopicName, new TopicMetadata(fooTopicId, fooTopicName, 2, mkMapOfPartitionRacks(2)));
+                        }
+                    })
+                    .withMember(new ConsumerGroupMember.Builder(memberId)
+                        .setInstanceId(instanceId)
+                        .setState(MemberState.STABLE)
+                        .setMemberEpoch(10)
+                        .setPreviousMemberEpoch(10)
+                        .setSubscribedTopicNames(Collections.singletonList(fooTopicName))
+                        .setAssignedPartitions(mkAssignment(
+                            mkTopicAssignment(fooTopicId, 0, 1)))
+                        .build())
+                    .withAssignment(memberId, mkAssignment(
+                        mkTopicAssignment(fooTopicId, 0, 1)))
+                    .withAssignmentEpoch(10))
+                .build();
+            context.groupMetadataManager.consumerGroup(groupId).setMetadataRefreshDeadline(Long.MAX_VALUE, 10);
+
+            JoinGroupRequestData request = new GroupMetadataManagerTestContext.JoinGroupRequestBuilder()
+                .withGroupId(groupId)
+                .withMemberId(UNKNOWN_MEMBER_ID)
+                .withGroupInstanceId(instanceId)
+                .withProtocols(GroupMetadataManagerTestContext.toConsumerProtocol(
+                    Collections.singletonList(fooTopicName),
+                    Collections.emptyList(),
+                    version))
+                .build();
+
+            // The static member joins with UNKNOWN_MEMBER_ID.
+            GroupMetadataManagerTestContext.JoinResult joinResult = context.sendClassicGroupJoin(
+                request,
+                true
+            );
+
+            if (replaySuccessfully) {
+                joinResult.appendFuture.complete(null);
+
+                String newMemberId = joinResult.joinFuture.get().memberId();
+                ConsumerGroupMember expectedMember = new ConsumerGroupMember.Builder(newMemberId)
+                    .setMemberEpoch(0)
+                    .setPreviousMemberEpoch(0)
+                    .setInstanceId(instanceId)
+                    .setState(MemberState.UNREVOKED_PARTITIONS)
+                    .setClientId("client")
+                    .setClientHost("localhost/127.0.0.1")
+                    .setSubscribedTopicNames(Collections.singletonList(fooTopicName))
+                    .setRebalanceTimeoutMs(500)
+                    .setSupportedClassicProtocols(request.protocols())
+                    .setPartitionsPendingRevocation(mkAssignment(
+                        mkTopicAssignment(fooTopicId, 0, 1)))
```

Comment:
This seems to be incorrect. When a static member rejoins, it should have the same assigned partitions. I think that the issue is that you don't provided an new assignment for the group.

Reply:
Yeah this is weird..😢 This is also because we don't know the new member id until join group is called.